src/Message/CompleteAuthoriseAndCaptureRequest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Omnipay\IcepayPayments\Message;
+
+use Omnipay\Common\Message\ResponseInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * The request for completing the authorise and capture at Icepay.
+ */
+class CompleteAuthoriseAndCaptureRequest extends AbstractRequest
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(): array
+    {
+        $data = parent::getData();
+
+        $data['ContractProfileId'] = $this->getContractProfileId();
+
+        return $data;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendData($data): ResponseInterface
+    {
+        $this->sendRequest(
+            Request::METHOD_POST,
+            sprintf(
+                '/transaction/%s',
+                $this->getTransactionReference()
+            ),
+            $data
+        );
+
+        $response = array_merge(
+            $this->getResponseBody(),
+            [
+                'statusCode' => $this->getResponse()->getStatusCode(),
```

Comment:
Yes, it should be `getCode()` because we expect the `ResponseInterface` of `Omnipay`. But sadly, that results a `null` result, therefore I use the `getStatusCode()` which results a 200 status code.